src/ride/water/water_coaster.c

```diff
@@ -61,13 +127,186 @@ static void paint_water_rc_track_flat(uint8 rideIndex, uint8 trackSequence, uint
+	case TRACK_ELEM_RIGHT_BANK_TO_25_DEG_DOWN:
+	case TRACK_ELEM_25_DEG_DOWN_TO_LEFT_BANK:
+	case TRACK_ELEM_25_DEG_DOWN_TO_RIGHT_BANK:
+	case TRACK_ELEM_LEFT_BANK:
+	case TRACK_ELEM_RIGHT_BANK:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_5_TILES_25_DEG_UP:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_5_TILES_25_DEG_UP:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_5_TILES_25_DEG_DOWN:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_5_TILES_25_DEG_DOWN:
+	case TRACK_ELEM_S_BEND_LEFT:
+	case TRACK_ELEM_S_BEND_RIGHT:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_3_TILES:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_3_TILES:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_3_TILES_BANK:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_3_TILES_BANK:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_3_TILES_25_DEG_UP:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_3_TILES_25_DEG_UP:
+	case TRACK_ELEM_LEFT_QUARTER_TURN_3_TILES_25_DEG_DOWN:
+	case TRACK_ELEM_RIGHT_QUARTER_TURN_3_TILES_25_DEG_DOWN:
+	case TRACK_ELEM_LEFT_HALF_BANKED_HELIX_UP_SMALL:
+	case TRACK_ELEM_RIGHT_HALF_BANKED_HELIX_UP_SMALL:
+	case TRACK_ELEM_LEFT_HALF_BANKED_HELIX_DOWN_SMALL:
+	case TRACK_ELEM_RIGHT_HALF_BANKED_HELIX_DOWN_SMALL:
+	case TRACK_ELEM_LEFT_HALF_BANKED_HELIX_UP_LARGE:
+	case TRACK_ELEM_RIGHT_HALF_BANKED_HELIX_UP_LARGE:
+	case TRACK_ELEM_LEFT_HALF_BANKED_HELIX_DOWN_LARGE:
+	case TRACK_ELEM_RIGHT_HALF_BANKED_HELIX_DOWN_LARGE:
+	case TRACK_ELEM_BRAKES:
+	case TRACK_ELEM_LEFT_EIGHTH_TO_DIAG:
+	case TRACK_ELEM_RIGHT_EIGHTH_TO_DIAG:
+	case TRACK_ELEM_LEFT_EIGHTH_TO_ORTHOGONAL:
+	case TRACK_ELEM_RIGHT_EIGHTH_TO_ORTHOGONAL:
+	case TRACK_ELEM_LEFT_EIGHTH_BANK_TO_DIAG:
+	case TRACK_ELEM_RIGHT_EIGHTH_BANK_TO_DIAG:
+	case TRACK_ELEM_LEFT_EIGHTH_BANK_TO_ORTHOGONAL:
+	case TRACK_ELEM_RIGHT_EIGHTH_BANK_TO_ORTHOGONAL:
+	case TRACK_ELEM_DIAG_FLAT:
+	case TRACK_ELEM_DIAG_25_DEG_UP:
+	case TRACK_ELEM_DIAG_60_DEG_UP:
+	case TRACK_ELEM_DIAG_FLAT_TO_25_DEG_UP:
+	case TRACK_ELEM_DIAG_25_DEG_UP_TO_60_DEG_UP:
+	case TRACK_ELEM_DIAG_60_DEG_UP_TO_25_DEG_UP:
+	case TRACK_ELEM_DIAG_25_DEG_UP_TO_FLAT:
+	case TRACK_ELEM_DIAG_25_DEG_DOWN:
+	case TRACK_ELEM_DIAG_60_DEG_DOWN:
+	case TRACK_ELEM_DIAG_FLAT_TO_25_DEG_DOWN:
+	case TRACK_ELEM_DIAG_25_DEG_DOWN_TO_60_DEG_DOWN:
+	case TRACK_ELEM_DIAG_60_DEG_DOWN_TO_25_DEG_DOWN:
+	case TRACK_ELEM_DIAG_25_DEG_DOWN_TO_FLAT:
+	case TRACK_ELEM_DIAG_FLAT_TO_LEFT_BANK:
+	case TRACK_ELEM_DIAG_FLAT_TO_RIGHT_BANK:
+	case TRACK_ELEM_DIAG_LEFT_BANK_TO_FLAT:
+	case TRACK_ELEM_DIAG_RIGHT_BANK_TO_FLAT:
+	case TRACK_ELEM_DIAG_LEFT_BANK_TO_25_DEG_UP:
+	case TRACK_ELEM_DIAG_RIGHT_BANK_TO_25_DEG_UP:
+	case TRACK_ELEM_DIAG_25_DEG_UP_TO_LEFT_BANK:
+	case TRACK_ELEM_DIAG_25_DEG_UP_TO_RIGHT_BANK:
+	case TRACK_ELEM_DIAG_LEFT_BANK_TO_25_DEG_DOWN:
+	case TRACK_ELEM_DIAG_RIGHT_BANK_TO_25_DEG_DOWN:
+	case TRACK_ELEM_DIAG_25_DEG_DOWN_TO_LEFT_BANK:
+	case TRACK_ELEM_DIAG_25_DEG_DOWN_TO_RIGHT_BANK:
+	case TRACK_ELEM_DIAG_LEFT_BANK:
+	case TRACK_ELEM_DIAG_RIGHT_BANK:
+	case TRACK_ELEM_BLOCK_BRAKES:
+		return get_track_paint_function_junior_rc(trackType, direction);
```

Comment:
You can also add the flat-to-60-degree pieces here, as there is drawing code for them too.